cookbooks/aws-parallelcluster-install/resources/lustre/lustre_centos7.rb

```diff
@@ -0,0 +1,122 @@
+# frozen_string_literal: true
+
+#
+#
+# Copyright:: 2013-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance with the
+# License. A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+# OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions and
+# limitations under the License.
+# Default resource implementation
+provides :lustre, platform: 'centos' do |node|
+  node['platform_version'].to_i == 7
+end
+
+use 'partial/_install_lustre_centos_redhat'
+
+lustre_version_hash = {
+  '7.6' => "2.10.6",
+  '7.5' => "2.10.5",
+}
+
+client_url_hash = {
+  '7.6' => "https://downloads.whamcloud.com/public/lustre/lustre-2.10.6/el7/client/RPMS/x86_64/lustre-client-2.10.6-1.el7.x86_64.rpm",
+  '7.5' => "https://downloads.whamcloud.com/public/lustre/lustre-2.10.5/el7.5.1804/client/RPMS/x86_64/lustre-client-2.10.5-1.el7.x86_64.rpm",
+}
+
+kmod_url_hash = {
+  '7.6' => "https://downloads.whamcloud.com/public/lustre/lustre-2.10.6/el7/client/RPMS/x86_64/kmod-lustre-client-2.10.6-1.el7.x86_64.rpm",
+  '7.5' => "https://downloads.whamcloud.com/public/lustre/lustre-2.10.5/el7.5.1804/client/RPMS/x86_64/kmod-lustre-client-2.10.5-1.el7.x86_64.rpm",
+}
+
+default_action :setup
+
+action :setup do
+  version = node['platform_version'].to_f
+  if %w(7.5 7.6).include?(version)
+    lustre_kmod_rpm = "#{node['cluster']['sources_dir']}/kmod-lustre-client-#{lustre_version_hash[version]}.x86_64.rpm"
+    lustre_client_rpm = "#{node['cluster']['sources_dir']}/lustre-client-#{lustre_version_hash[version]}.x86_64.rpm"
+
+    # Get Lustre Kernel Module RPM
+    remote_file lustre_kmod_rpm do
+      source kmod_url_hash[version]
+      mode '0644'
+      retries 3
+      retry_delay 5
+      action :create_if_missing
+    end
+
+    # Get Lustre Client RPM
+    remote_file lustre_client_rpm do
+      source client_url_hash[version]
+      mode '0644'
+      retries 3
+      retry_delay 5
+      action :create_if_missing
+    end
+
+    # Install lustre mount drivers
+    package 'lustre_kmod' do
+      source lustre_kmod_rpm
+    end
+
+    # Install lustre mount drivers
+    package 'lustre_client' do
+      source lustre_client_rpm
+    end
+
+    kernel_module 'lnet'
+
+  elsif node['platform_version'].to_f >= 7.7
+    execute 'yum-config-manager_skip_if_unavail' do
+      command "yum-config-manager --setopt=\*.skip_if_unavailable=1 --save"
+    end
+    action_install_lustre
```

Comment:
Just out of curiosity again: did you check packages for `5.5` and `5.6` are still not available in the repository?

Reply:
I guess you mean `7.5` and `7.6`
I haven't done an explicit check but also in the public documentation https://docs.aws.amazon.com/fsx/latest/LustreGuide/install-lustre-client.html we can see a different path for downloading the needed packages in those versions of Centos

Reply:
Sure, I meant `7.x`.
Thank you for the clarification.